glue/sales_val_flagging.py

```diff
@@ -152,30 +144,98 @@ def group_size_adjustment(df, stat_groups: list, min_threshold, condos: bool):
         df: dataframe with newly manually adjusted outlier values.
 
     """
+
+    # - - -
+    # Handle group threshold adjustment
+    # TODO: Explain choice of sv_outlier_reason$n, do we set all reasons to 0 as well?
+    # - - -
     group_counts = df.groupby(stat_groups).size().reset_index(name="count")
     filtered_groups = group_counts[group_counts["count"] <= min_threshold]
 
     # Merge df_flagged with filtered_groups on the columns to get the matching rows
-    merged_df = pd.merge(
+    df = pd.merge(
         df, filtered_groups[stat_groups], on=stat_groups, how="left", indicator=True
     )
 
-    # Modify the .loc condition to include checking for sv_outlier_type values
-    condition = (merged_df["_merge"] == "both") & (merged_df["sv_is_outlier"] == 1)
+    # Assign sv_outlier_reasons
+    for idx in range(1, 4):
+        df[f"sv_outlier_reason{idx}"] = np.nan
+
+    outlier_type_crosswalk = {
+        "sv_ind_ptax_flag_w_deviation": "PTAX-203 Exclusion",
+        "sv_ind_price_high_price": "High price",
+        "sv_ind_price_low_price": "Low price",
+        "sv_ind_price_high_price_sqft": "High price per square foot",
+        "sv_ind_price_low_price_sqft": "Low price per square foot",
+        "sv_ind_char_short_term_owner": "Short-term owner",
+        "sv_ind_char_family_sale": "Family Sale",
+        "sv_ind_char_non_person_sale": "Non-person sale",
+        "sv_ind_char_statistical_anomaly": "Statistical Anomaly",
+        "sv_ind_char_price_swing_homeflip": "Price swing / Home flip",
+    }
+
+    group_thresh_price_fix = [
+        "sv_ind_price_high_price"
+        "sv_ind_price_low_price"
+        "sv_ind_price_high_price_sqft"
+        "sv_ind_price_low_price_sqft"
+    ]
 
-    # Using .loc[] to set the desired values for rows meeting the condition
-    merged_df.loc[condition, "sv_outlier_type"] = "Not outlier"
-    merged_df.loc[condition, "sv_is_outlier"] = 0
+    def fill_outlier_reasons(row):
+        reason_idx = 1
+        for reason in outlier_type_crosswalk:
+            # Add a check to ensure that only specific reasons are added if _merge is not 'both'
+            if (
+                reason in row
+                and row[reason]
+                and not (row["_merge"] == "both" and reason in group_thresh_price_fix)
+            ):
+                row[f"sv_outlier_reason{reason_idx}"] = outlier_type_crosswalk[reason]
+                if reason_idx >= 3:
+                    break
+                reason_idx += 1
+        return row
+
+    df = df.apply(fill_outlier_reasons, axis=1)
```

Comment:
This is how we I handled the implementation of not counting the price outliers but retaining the char outliers given a group threshold violation.

The `row["_merge"] == "both"` check checks for the group thresh violation, and the `reason in group_thresh_price_fix` checks to see if we are looking at a price indicator column. So it basically says:

if not ( below threshold and price outlier) skip this iteration in the loop, which means the condition won't be filled in. 

I thought this approach would be better than manually re-arranging at the end of the function.